### PR TITLE
 Hide N/A and null values in event log #8376

### LINKED ISSUE #8376: Hide N/A and Null values in events log on frontend
image

### PR DESCRIPTION
**Pull request by Nithin9585**
## Proposed Changes

- Fixes #8376
- Hide `N/A` and `null` values in the event log on the frontend.
- Updated the `NeurologicalTables` component to conditionally render empty strings for `N/A` and `null` values.

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [X] Add specs that demonstrate bug / test a new feature.
- [X] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [X] Ensure that UI text is kept in I18n files.
- [X] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

File: src/Components/Facility/Consultations/NeurologicalTables.tsx
Comment (rithviknishad) [inline src/Components/Facility/Consultations/NeurologicalTables.tsx:1]: This file has nothing to do with Event Logs.
You might want to have a look at this file instead: https://github.com/coronasafe/care_fe/blob/develop/src/Components/Facility/ConsultationDetails/Events/GenericEvent.tsx

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/Components/Facility/ConsultationDetails/Events/GenericEvent.tsx [hunk 1]
```diff
@@ -88,16 +88,18 @@ export default function GenericEvent(props: IProps) {
...
+        .map(([key, value]) => (
+          <div key={key} className="flex w-full flex-col items-start gap-2">
+            <span className="text-xs capitalize text-secondary-700">
+              {t(key).replaceAll(/_/g, " ")}
+            </span>
+            <span className="break-all text-sm font-semibold text-secondary-700">
+              {formatValue(value, key)}
+            </span>
+          </div>
+        ))}
```
Comment (rithviknishad) [inline src/Components/Facility/ConsultationDetails/Events/GenericEvent.tsx]: LGTM, but can we map with the formatted value before the filter? `formatValue` function for the same entry is now called twice, which could be avoided.

Something like:

```jsx
Object.entries(values)
  .map(([key, value]) => ({ key, value: formatValue(value, key)}))
  .filter(({ value }) => value != null)
  .map(({ key, value }) => <>...</>)
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/Components/Facility/Consultations/NeurologicalTables.tsx
Comment (rithviknishad) [inline src/Components/Facility/Consultations/NeurologicalTables.tsx:1]: Could you discard/undo the changes made to the neurological tables file?

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Review by rithviknishad — APPROVED** [review]

**Review by shivankacker — APPROVED** [review]

File: src/Components/Facility/ConsultationDetails/Events/GenericEvent.tsx [hunk 2]
```diff
@@ -50,8 +50,8 @@ const formatValue = (value: unknown, key?: string): ReactNode => {
 
       return (
         <ul className="list-disc space-y-2 pl-4">
-          {value.map((v) => (
-            <li>{formatValue(v, key)}</li>
+          {value.map((v, idx) => (
+            <li key={idx}>{formatValue(v, key)}</li>
```
Comment (rithviknishad) [inline src/Components/Facility/ConsultationDetails/Events/GenericEvent.tsx:54]: Nice find!

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**